kubernetes-client/src/main/java/io/fabric8/kubernetes/client/utils/HttpClientUtils.java

```diff
@@ -18,167 +18,184 @@
-            }
+      if (config.isTrustCerts()) {
+        httpClientBuilder.hostnameVerifier(new HostnameVerifier() {
+          @Override
+          public boolean verify(String s, SSLSession sslSession) {
+            return true;
+          }
+        });
+      }
+
+      TrustManager[] trustManagers = SSLUtils.trustManagers(config);
+      KeyManager[] keyManagers = SSLUtils.keyManagers(config);
+
+      if (keyManagers != null || trustManagers != null || config.isTrustCerts()) {
+        X509TrustManager trustManager = null;
+        if (trustManagers != null && trustManagers.length == 1) {
+          trustManager = (X509TrustManager) trustManagers[0];
+        }
 
-          httpClientBuilder.addInterceptor(new Interceptor() {
-            @Override
-            public Response intercept(Chain chain) throws IOException {
-              Request request = chain.request();
+        try {
+          SSLContext sslContext = SSLUtils.sslContext(keyManagers, trustManagers, config.isTrustCerts());
+          httpClientBuilder.sslSocketFactory(sslContext.getSocketFactory(), trustManager);
+        } catch (GeneralSecurityException e) {
+          throw new AssertionError(); // The system has no TLS. Just give up.
+        }
+      } else {
+        SSLContext context = SSLContext.getInstance("TLSv1.2");
+        context.init(keyManagers, trustManagers, null);
+        httpClientBuilder.sslSocketFactory(context.getSocketFactory(), (X509TrustManager) trustManagers[0]);
+      }
+
+      httpClientBuilder.addInterceptor(new Interceptor() {
+        @Override
+        public Response intercept(Chain chain) throws IOException {
+          final int maxRetries = config.getMaxRetries();
+          final int retryDelay = config.getRetryDelayInSeconds();
+          Response response = null;
+          int retries = 0;
+          do {
+            try {
               if (isNotNullOrEmpty(config.getUsername()) && isNotNullOrEmpty(config.getPassword())) {
+                LOGGER.info("Using username and password Authorization");
                 Request authReq = chain.request().newBuilder().addHeader("Authorization", Credentials.basic(config.getUsername(), config.getPassword())).build();
-                return chain.proceed(authReq);
+                response = chain.proceed(authReq);
               } else if (isNotNullOrEmpty(config.getOauthToken())) {
+                LOGGER.info("Using OAuth Authorization");
                 Request authReq = chain.request().newBuilder().addHeader("Authorization", "Bearer " + config.getOauthToken()).build();
-                return chain.proceed(authReq);
+                response = chain.proceed(authReq);
+              }
+            } catch (ConnectException e) {
+              LOGGER.warn("Request failed retry {} ", retries);
+              try {
+                SECONDS.sleep(retryDelay);
+              } catch (InterruptedException e1) {
+                //dont worry
```

Comment:
We shouldn't ignore the fact that the thread gets interrupted.
Since the method signature does throw IOEXception, I would throw java.io.InterruptedIOException to signal that the interceptor was interrupted.